### PR TITLE
docs: improved sql/dql/join in en

### PR DESCRIPTION
**Pull request by michelle-qinqin**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
change splice=>join or concatenation 
added sql examples
updated the boundary table

fixed format error on line 9 in zh version

### CODE REVIEW

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 1]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+|:-----------------------------------------------------------|--------------|---------------------|---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| JOIN Clause                | **``✓``**    | **``✓``**           | **``✓``**           | The Join clause indicates that the data source comes from multiple joined tables. OpenMLDB currently only supports LAST JOIN. During Online Serving, please follow [the specification of LAST JOIN under Online Serving](../deployment_manage/ONLINE_SERVING_REQUIREMENTS.md#online-servinglast-join)                                                      |
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: During Online Serving
This actually means the "Online Request Mode"? We'd better keep them consistent.

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 2]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
 
-```sql
--- desc: simple spelling query without ORDER BY
 
-SELECT t1.col1 as t1_col1, t2.col1 as t2_col2 from t1 LAST JOIN t2 ON t1.col1 = t2.col1
-```
+### LAST JOIN without ORDER BY
 
-When `LAST JOIN` is spliced ​​without sorting, the first hit data row is spliced
+#### Example1: Unsorted LAST JOIN
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: maybe no need this title?

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 3]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+
+The unsorted `LAST JOIN` will concat every row of the left table with the last matching row of the right table.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: matching => matched

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 4]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
 
 
-Take the second row of the left table as an example, the right table that meets the conditions is unordered, there are 2 hit conditions, select the last one `5, b, 2020-05-20 10:11:12`
+Take the second row of the left table as an example. The right table is unordered, and there are 2 matching rows. The last one `5, b, 2020-05-20 10:11:12` will be joined with the second row of the left.  
 
 ![Figure 8: last join without order result](../dql/images/last_join_without_order2.png)
 
 The final result is shown in the figure above.
 
+##### SQL Example using OpenMLDB CLI
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: I think no need to make any examples as a separate section
=>

**Example of LAST JOIN without ORDER_BY**

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 5]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+The following SQL commands created the left table t1 as mentioned above and inserted corresponding data.
+In order to check the results conveniently, it is recommended to create index on `col1` and use `std_ts` as timestamp. It doesn't matter if you create t1 without index, since it doesn't influence the concatenation in this case.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: influence => affect

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 6]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+  2    b      20200520101114
+  3    c      20200520101116
+ ---- ------ ----------------
+
+3 rows in set
+```      
+The following SQL commands created the right table t2 as mentioned above and inserted corresponding data.
+
+```{note}
+The storage order of data rows is not necessarily the same as their insert order. And the storage order will influence the matching order when JOIN.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md:79]: The physical storage order of rows is not necessarily the same as their insertion order. However, the storage order will affect the matching order when performing the LAST JOIN.

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 7]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+In this example, we want to realize the storage order of t2 as the above figure displayed, which will lead to a result that is convenient to check.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md:80]: realize => implement

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 8]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+  3    c      20200520101116   6    c      20200520101113
+  1    a      20200520101112   2    a      20200520101113
+  2    b      20200520101114   5    b      20200520101112
+ ---- ------ ---------------- ---- ------ ----------------
+
+3 rows in set
+```
+
+```{note}
+The execution of `LAST JOIN` can be optimized by index. If there is index corresponding with the `order by` and conditions in `LAST JOIN` clause, its `ts` will be used as the implicit order for unsorted `LAST JOIN`. If there is not index like this, the implicit order is the storage order. But the storage order of a table without index is unpredictable.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: corresponding with the `order by`
=>
corresponding to the `ORDER BY`

Comment (lumianph) [reply docs/en/reference/sql/dql/JOIN_CLAUSE.md]: If there is not index
If there is no index

Comment (lumianph) [reply docs/en/reference/sql/dql/JOIN_CLAUSE.md]: unpredictable => undefined.

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 9]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+If the `ts` was not given when create index, OpenMLDB uses the time when the data was inserted as `ts`.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md:142]: create => creating
uses the time => uses the timestamp

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 10]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+The execution of `LAST JOIN` can be optimized by index. If there is index corresponding with the `order by` and conditions in `LAST JOIN` clause, its `ts` will be used as the implicit order for unsorted `LAST JOIN`. If there is not index like this, the implicit order is the storage order. But the storage order of a table without index is unpredictable.
+If the `ts` was not given when create index, OpenMLDB uses the time when the data was inserted as `ts`. 
+```
+
+
+
 ### LAST JOIN with ORDER BY
 
-#### Example: LAST JOIN Sorting And Splicing
+#### Example2: Sorted LAST JOIN
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: remove this title as well?

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 11]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
 
-```SQL
--- desc: Simple spelling query with ORDER BY
-SELECT t1.col1 as t1_col1, t2.col1 as t2_col2 from t1 LAST JOIN t2 ORDER BY t2.std_ts ON t1.col1 = t2.col1
-```
 
-When `LAST JOIN` is configured with `Order By`, the right table is sorted by Order, and the last hit data row is spliced.
+When `LAST JOIN` is configured with `Order By`, the right table is sorted by `Order`, and the last matching data row will be joined.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: ORDER BY
the right table is sorted by the specified order
matching data row => matched row

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 12]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
 
 ![Figure 9: last join with order](../dql/images/last_join_with_order1.png)
 
-Taking the second row of the left table as an example, there are 2 items in the right table that meet the conditions. After sorting by `std_ts`, select the last item `3, b, 2020-05-20 10:11:13`
+Taking the second row of the left table as an example, there are 2 items in the right table that meet the conditions. After sorting by `std_ts`, the last row `3, b, 2020-05-20 10:11:13` will be joined.
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: 2 items => two rows

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 13]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
 
 ![Figure 10: last join with order result](../dql/images/last_join_with_order2.png)
 
 The final result is shown in the figure above.
+
+##### SQL Example using OpenMLDB CLI
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: **Example of LAST JOIN with ORDER BY**

File: docs/en/reference/sql/dql/JOIN_CLAUSE.md [hunk 14]
```diff
@@ -18,51 +19,219 @@ JoinType ::= 'LAST'
...
+ ---- ------ ---------------- ---- ------ ----------------
+  id   col1   std_ts           id   col1   std_ts
+ ---- ------ ---------------- ---- ------ ----------------
+  1    a      20200520101112   2    a      20200520101113
+  2    b      20200520101114   3    b      20200520101113
+  3    c      20200520101116   4    c      20200520101114
+ ---- ------ ---------------- ---- ------ ----------------
+```
+
+#### Example3: LAST JOIN without Matching Row
```
Comment (lumianph) [inline docs/en/reference/sql/dql/JOIN_CLAUSE.md]: **LAST JOIN with No Matched Rows**

File: docs/zh/reference/sql/dql/JOIN_CLAUSE.md [hunk 15]
```diff
@@ -19,18 +19,19 @@ JoinType ::= 'LAST'
...
-| :------------- | --------------- | :----------------------------------------------------------- |
-| JOIN Clause    | 仅支持LAST JOIN | 表示数据来源多个表JOIN。OpenMLDB目前仅支持LAST JOIN。在Online Serving时，需要遵循[Online Serving下LAST JOIN的使用规范](../deployment_manage/ONLINE_SERVING_REQUIREMENTS.md#online-serving下last-join的使用规范) |
+| SELECT语句元素                                 | 离线模式  | 在线预览模式 | 在线请求模式 | 说明                                                                                                                                                                                                                              |
+| :--------------------------------------------- | --------- | ------------ | ------------ |:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| JOIN Clause| **``✓``** | **``✓``** | **``✓``** | 表示数据来源多个表JOIN。OpenMLDB目前仅支持LAST JOIN。在Online Serving时，需要遵循[Online Serving下LAST JOIN的使用规范](https://openmldb.ai/docs/zh/main/reference/sql/deployment_manage/ONLINE_SERVING_REQUIREMENTS.html#online-servinglast-join)|
```
Comment (lumianph) [inline docs/zh/reference/sql/dql/JOIN_CLAUSE.md]: online serving 是否就是在线请求模式，保持一致

**Review by lumianph — CHANGES_REQUESTED** [review]
可能有些格式（比如example不要作为title formatting），中文文档你再检查一下

File: docs/zh/reference/sql/dql/JOIN_CLAUSE.md (hunk 15 above)
Comment (michelle-qinqin) [reply docs/zh/reference/sql/dql/JOIN_CLAUSE.md]: 事实上Online serving下的使用规范在离线模式下也是需要遵守的。不仅是join，window也是如此。
根据之前的实践，在单机版离线模式下，假如不遵循实践规范，虽然语句能跑通（不违反语法），但是运行结果很可能达不到目的。
但是这些信息并没有在JOIN 和 WINDOW 子句单独的文档中提及。
如果需要修改，感觉[sql上线规范和要求](https://openmldb.ai/docs/zh/main/reference/sql/deployment_manage/ONLINE_SERVING_REQUIREMENTS.html)也需要大量改动？

**Comment by michelle-qinqin** [comment]
> 

中英文文档的格式均已修改。

**Comment by michelle-qinqin** [comment]
所有的 matching rows 均已修改为mtached rows

**Comment by michelle-qinqin** [comment]
中英文文档的JOIN、WINDOW和中文的SELECT概述的Online Serving均已修改。
英文的SELECT概述在另一个PR中修改：[#229](https://github.com/4paradigm/OpenMLDB/pull/2296)

**Review by lumianph — APPROVED** [review]

**Pull request merged**